BSPConversionCmd/Program.cs

```diff
@@ -22,39 +22,62 @@ class Options
-			//args[1] = @"c:\users\tyler\documents\tools\source engine\bspconvert\output";
-			//args[2] = "--newbsp";
-
-			Parser.Default.ParseArguments<Options>(args)
-				.WithParsed(options =>
-				{
-					if (options.DisplacementPower < 2 || options.DisplacementPower > 4)
-						throw new ArgumentOutOfRangeException("Displacement power must be between 2 and 4.");
-
-					var converterOptions = new BSPConverterOptions()
-					{
-						noPak = options.NoPak,
-						skyFix = options.SkyFix,
-						DisplacementPower = options.DisplacementPower,
-						newBSP = options.NewBSP,
-						prefix = options.Prefix,
-						inputFile = options.InputFile,
-						outputDir = options.OutputDirectory
-					};
-					var converter = new BSPConverter(converterOptions, new ConsoleLogger());
-					converter.Convert();
-				});
-		}
+            //args = new string[3];
+            //args[0] = @"c:\users\tyler\documents\tools\source engine\bspconvert\kopo_fe.pk3";
+            //args[1] = @"c:\users\tyler\documents\tools\source engine\bspconvert\output";
+            //args[2] = "--newbsp";
+
+            string[] inputFilePaths = new string[] { };
+            List<string> inputEntries = new List<string>();
+            foreach (var entry in args)
+            {
+                //Console.WriteLine(entry);
+                if (Path.HasExtension(entry))
+                {
+                    inputEntries.Add(entry);
+                    inputFilePaths = inputEntries.ToArray();
+                }
+            }
+            string[] splitArgs = (String.Join(",", inputFilePaths).Split(','));
+
+            Parser.Default.ParseArguments<Options>(args)
+            .WithParsed(options =>
+            {
+                if (options.DisplacementPower < 2 || options.DisplacementPower > 4)
+                    throw new ArgumentOutOfRangeException("Displacement power must be between 2 and 4.");
+
+                if (splitArgs.Length > 0)
+                    Console.WriteLine(@"Converting... (may take more than a few seconds)");
+
+                foreach (var entry in splitArgs)
+                {
+                    options.InputFile = entry;
+                    var converterOptions = new BSPConverterOptions()
+                    {
+                        noPak = options.NoPak,
+                        skyFix = options.SkyFix,
+                        DisplacementPower = options.DisplacementPower,
+                        newBSP = options.NewBSP,
+                        prefix = options.Prefix,
+                        inputFile = options.InputFile,
+                        outputDir = options.OutputDirectory
+                    };
+                    var converter = new BSPConverter(converterOptions, new ConsoleLogger());
+                    converter.Convert();
+                }
+
+                Console.WriteLine($"\n" + ((splitArgs.Length > 0) ? "Convert Finished! " : "") + "Press any key to Exit.");
```

Comment:
This is not needed, the command should just complete after finishing.

Reply:
There is a log message that displays the converted BSP's path when the conversion is finished.